api/s2n.h

```diff
@@ -687,6 +687,266 @@ extern int s2n_cert_get_utf8_string_from_extension_data_length(const uint8_t *ex
+ * greater than this `max_identity_length` value an error will be returned.
+ */
+S2N_API 
+int s2n_connection_get_negotiated_psk_identity(struct s2n_connection *conn, uint8_t *identity, uint16_t max_identity_length);
+
+struct s2n_offered_psk;
+
+/**
+ * Creates a new s2n offered PSK object. 
+ * An offered PSK object represents a single PSK sent by the client.
+ * 
+ * # Safety
+ * 
+ * Use `s2n_offered_psk_free` to free the memory allocated to the s2n offered PSK object created by this API. 
+ *
+ * @return struct s2n_offered_psk* Returns a pointer to the newly created offered PSK object.
+ */
+S2N_API 
+struct s2n_offered_psk* s2n_offered_psk_new();
+
+/**
+ * Frees the memory associated with the `s2n_offered_psk` object.
+ *
+ * @param psk A pointer to the `s2n_offered_psk` object to be freed.
+ */
+S2N_API 
+int s2n_offered_psk_free(struct s2n_offered_psk **psk);
+
+/**
+ * Gets the PSK identity and PSK identity length for a given offered PSK object. 
+ * 
+ * @param psk A pointer to the offered PSK object being read.
+ * @param identity The PSK identity being obtained.
+ * @param size The length of the PSK identity being obtained.
+ */
+S2N_API 
+int s2n_offered_psk_get_identity(struct s2n_offered_psk *psk, uint8_t** identity, uint16_t *size);
+
+struct s2n_offered_psk_list;
+
+/**
+ * Checks whether the offered PSK list has an offered psk object next in line in the list.
+ * An offered PSK list contains all the PSKs offered by the client for the server to select.
+ * 
+ * @param psk_list A pointer to the offered PSK list being read.
+ * @return bool A boolean value representing whether an offered psk object is present next in line in the offered PSK list.
+ */
+S2N_API 
+bool s2n_offered_psk_list_has_next(struct s2n_offered_psk_list *psk_list);
+
+/**
+ * Obtains the next offered PSK object from the list of offered PSKs. Use `s2n_offered_psk_list_has_next` 
+ * prior to this API call to ensure we have not reached the end of the list.
+ * 
+ * @param psk_list A pointer to the offered PSK list being read.
+ * @param psk A pointer to the next offered PSK object being obtained.
+ */
+S2N_API 
+int s2n_offered_psk_list_next(struct s2n_offered_psk_list *psk_list, struct s2n_offered_psk *psk);
+
+/**
+ * Returns the offered PSK list to its original read state.
+ *
+ * When `s2n_offered_psk_list_reread` is called, `s2n_offered_psk_list_next` will return the first PSK 
+ * in the offered PSK list.
+ *
+ * @param psk_list A pointer to the offered PSK list being reread.
+ */
+S2N_API 
+int s2n_offered_psk_list_reread(struct s2n_offered_psk_list *psk_list);
+
+/**
+ * Chooses a PSK from the offered PSK list to be used for the connection.  
+ * This API matches the PSK identity received from the client against the server's known PSK identities 
+ * list, in order to choose the PSK to be used for the connection. If the PSK identity sent from the client 
```

Comment:
I still don't think you're communicating what you want by describing how this API is implemented. If you are trying to imply that you need to make sure the PSK you choose is present on the server connection and should be added with s2n_connection_append_psk if it's not, then say that explicitly.